File: VB/Default MVVM/ColumnBehavior/ColumnBehavior.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Windows
Imports System.Windows.Controls
Imports DevExpress.Xpf.Grid

Namespace Default_MVVM
	Public NotInheritable Class ColumnBehavior
		Public Shared ReadOnly IsRenameEditorActivatedProperty As DependencyProperty = DependencyProperty.RegisterAttached("IsRenameEditorActivated", GetType(Boolean), GetType(ColumnBehavior), New PropertyMetadata(False))

		Private Sub New()
		End Sub
		Public Shared Sub SetIsRenameEditorActivated(ByVal element As GridColumn, ByVal value As Boolean)
			If element.Header Is Nothing AndAlso value Then
				element.Header = element.FieldName
			End If
			element.SetValue(IsRenameEditorActivatedProperty, value)
		End Sub
		Public Shared Function GetIsRenameEditorActivated(ByVal element As GridColumn) As Boolean
			Return CBool(element.GetValue(IsRenameEditorActivatedProperty))
		End Function
	End Class
End Namespace

File: CS/Default MVVM/ColumnBehavior/ColumnBehavior.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Windows;
using System.Windows.Controls;
using DevExpress.Xpf.Grid;

namespace Default_MVVM
{
    public static class ColumnBehavior
    {
        public static readonly DependencyProperty IsRenameEditorActivatedProperty = DependencyProperty.RegisterAttached("IsRenameEditorActivated",
                                                                                        typeof(bool),
                                                                                        typeof(ColumnBehavior),
                                                                                        new PropertyMetadata(false));

        public static void SetIsRenameEditorActivated(GridColumn element, bool value)
        {
            if (element.Header == null && value)
                element.Header = element.FieldName;
            element.SetValue(IsRenameEditorActivatedProperty, value);
        }
        public static bool GetIsRenameEditorActivated(GridColumn element)
        {
            return (bool)element.GetValue(IsRenameEditorActivatedProperty);
        }
    }
}